### ISSUE TITLE
debug: show errors when remote connection fails

### ISSUE THREAD

**Issue by vincentjocodes** [golang/vscode-go#215]
### What version of Go, VS Code & VS Code Go extension are you using?
- Run `go version` to get version of Go
	- go version go1.14.3 darwin/amd64
- Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders
	- 1.46.0
- Check your installed extensions to get the version of the VS Code Go extension 
	- v0.14.4

### Describe the bug
`json-rpc2` module doesn't catch all the errors in its callback in this line of code: https://github.com/golang/vscode-go/blob/master/src/debugAdapter/goDebug.ts#L597. 

Basically if a remote connection fails, it won't call the callback for errors. So the connection hangs when we use vscode-go debugger. There is no future plan to handle this case: https://github.com/pocesar/node-jsonrpc2/issues/53. 

One way I've experimented and had success with is something like:

```
client.on('error', (error) => {
     reject(error);
});
```

**Comment by stamblerre** [comment]
Thanks for this report and investigation, @vincentjocodes! Feel free to [contribute](https://github.com/golang/vscode-go/blob/master/docs/contributing.md) your patch, if you're interested in doing so.

**Comment by hyangah** [comment]
There is an issue for replacing json-rpc2 https://github.com/golang/vscode-go/issues/113
Will the new library surface the underlying connection errors better?

**Comment by vincentjocodes** [comment]
@stamblerre Just did a patch, please take a look. https://github.com/golang/vscode-go/pull/216. Thanks!
